### PR TITLE
fix: LT-2334 - Persist queue ID across case pages

### PR DESCRIPTION
**Pull request by jfaracik**
* Move queue cases page into queues app for separation between queues and cases
* Use reverse urls in cases page
* Cases page now uses built in pagination class

### CODE REVIEW

File: cases/urls.py [hunk 1]
```diff
@@ -4,127 +4,113 @@
...
-    path("<uuid:pk>/team-advice-view/coalesce/", advice.CoalesceUserAdvice.as_view(), name="coalesce_user_advice"),
-    path("<uuid:pk>/team-advice-view/", advice.ViewTeamAdvice.as_view(), name="team_advice_view"),
-    path("<uuid:pk>/final-advice-view/coalesce/", advice.CoalesceTeamAdvice.as_view(), name="coalesce_team_advice"),
-    path("<uuid:pk>/final-advice-view/", advice.ViewFinalAdvice.as_view(), name="final_advice_view"),
-    path("<uuid:pk>/advice-view/give-user-advice/", advice.GiveUserAdvice.as_view(), name="give_user_advice"),
-    path(
-        "<uuid:pk>/advice-view/give-user-advice/<str:type>/",
+    path("", main.ViewCase.as_view(), name="case"),
+    path("done/", main.CaseProcessedByUser.as_view(), name="done"),
+    path("done/<uuid:queue_id>", main.CaseProcessedByUserForQueue.as_view(), name="done_for_queue"),
```
Comment (BenAndrew-Kainos) [inline cases/urls.py]: As the queue ID is now persistent we will need to merge these 2 views

File: conf/context_processors.py [hunk 2]
```diff
@@ -7,9 +7,24 @@
...
+    if "queue_pk" not in request.resolver_match.kwargs:
+        return {}
+
+    queue_pk = request.resolver_match.kwargs["queue_pk"]
+    queue = get_queue(request, queue_pk)
+
+    print("\n")
+    print("queue")
+    print(queue)
+    print("\n")
```
Comment (BenAndrew-Kainos) [inline conf/context_processors.py]: Remember to remove in the final commit

Comment (BenAndrew-Kainos) [inline conf/context_processors.py]: Remember to remove in the final commit

File: conf/context_processors.py [hunk 3]
```diff
@@ -7,9 +7,24 @@
...
+
+    return {"queue": {"id": queue_pk, "name": queue["name"], "is_system_queue": queue["is_system_queue"]}}
```
Comment (BenAndrew-Kainos) [inline conf/context_processors.py]: Is this necessary (can we not just. use get_queue which already returns this format)

File: conf/context_processors.py [hunk 4]
```diff
@@ -23,15 +38,15 @@ def export_vars(request):
...
+            {"title": "Cases", "url": reverse_lazy("core:index"), "icon": "menu/cases"},
             {
                 "title": OrganisationsPage.TITLE,
                 "url": reverse_lazy("organisations:organisations"),
                 "icon": "menu/businesses",
             },
             {"title": "Teams", "url": reverse_lazy("teams:teams"), "icon": "menu/teams"},
             {"title": "My Team", "url": reverse_lazy("teams:team"), "icon": "menu/teams"},
-            {"title": QueuesList.TITLE, "url": reverse_lazy("queues:queues"), "icon": "menu/queues"},
+            # {"title": QueuesList.TITLE, "url": reverse_lazy("queues:queues"), "icon": "menu/queues"},
```
Comment (BenAndrew-Kainos) [inline conf/context_processors.py]: Why has this been commented out?

File: core/urls.py [hunk 5]
```diff
@@ -5,6 +5,6 @@
 app_name = "core"
 
 urlpatterns = [
-    path("", views.hub, name="hub"),
+    path("", views.Index.as_view(), name="index"),
```
Comment (BenAndrew-Kainos) [inline core/urls.py]: Nice :)

File: core/views.py [hunk 6]
```diff
@@ -1,8 +1,10 @@
-from django.shortcuts import render, redirect
+from django.shortcuts import render
+from django.urls import reverse_lazy
+from django.views.generic import RedirectView
 
 
-def hub(request):
-    return redirect("/cases/")
+class Index(RedirectView):
+    url = reverse_lazy("queues:cases")
```
Comment (BenAndrew-Kainos) [inline core/views.py]: Should we not just put the index page class here

File: conf/context_processors.py (hunk 2 above)
Comment (jfaracik) [reply conf/context_processors.py]: Done!

File: conf/context_processors.py (hunk 3 above)
Comment (jfaracik) [reply conf/context_processors.py]: Done!

File: conf/context_processors.py (hunk 4 above)
Comment (jfaracik) [reply conf/context_processors.py]: Uncommented again (was just for development)

File: core/views.py (hunk 6 above)
Comment (jfaracik) [reply core/views.py]: Definitely an idea, I just wanted to keep queues grouped with queues, cases grouped with cases.

File: conf/context_processors.py (hunk 4 above)
Comment (BenAndrew-Kainos) [reply conf/context_processors.py]: 👍

File: core/views.py (hunk 6 above)
Comment (BenAndrew-Kainos) [reply core/views.py]: Sure that makes sense, can we just point the URL to that view then or does it need to be a redirect for something to do with queues?

Comment (jfaracik) [reply core/views.py]: Yeah can just point it :)

Comment (jfaracik) [reply core/views.py]: Done!

File: core/views.py [hunk 7]
```diff
@@ -1,8 +1,4 @@
-from django.shortcuts import render, redirect
-
-
-def hub(request):
-    return redirect("/cases/")
+from django.shortcuts import render
```
Comment (BenAndrew-Kainos) [inline core/views.py]: Do we still need this file?

Comment (jfaracik) [reply core/views.py]: Sadly, used for the menu fallback for non-JS users.

**Pull request merged**